server/web/httpserver.go

```diff
@@ -1,16 +1,52 @@
 package web
 
 import (
+	"context"
+	"encoding/json"
+	"errors"
+	gw "github.com/TUM-Dev/Campus-Backend/api"
+	"github.com/TUM-Dev/Campus-Backend/backend"
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
 	"net"
 	"net/http"
 )
 
-func HTTPServe(l net.Listener) error {
-	mux := http.NewServeMux()
-	mux.HandleFunc("/", func(w http.ResponseWriter, _ *http.Request) {
-		w.Write([]byte("Welcome to the new campus API."))
-	})
+func HTTPServe(l net.Listener, grpcPort string) error {
+	mux := runtime.NewServeMux(runtime.WithIncomingHeaderMatcher(
+		func(s string) (string, bool) {
+			return s, true
+		}),
+		runtime.WithErrorHandler(errorHandler),
+	)
+	opts := []grpc.DialOption{grpc.WithInsecure()}
+	err := gw.RegisterCampusHandlerFromEndpoint(context.TODO(), mux, grpcPort, opts)
+	if err != nil {
+		return err
+	}
 
 	s := &http.Server{Handler: mux}
 	return s.Serve(l)
 }
+
+// errorHandler translates gRPC raised by the backend into HTTP errors.
+func errorHandler(ctx context.Context, mux *runtime.ServeMux, marshaler runtime.Marshaler, w http.ResponseWriter, _ *http.Request, err error) {
+	httpStatus := http.StatusInternalServerError
+	httpResponse := "Internal Server Error"
+	if errors.Is(err, backend.ErrNoDeviceID) {
+		httpStatus = http.StatusForbidden
+		httpResponse = "Not Authorized"
+	}
+	w.WriteHeader(httpStatus)
+	// Marshal won't fail, we know all inputs.
```

Comment:
That's a bold strategy cotton!
![image](https://user-images.githubusercontent.com/298860/147082254-ebf2161e-f601-4664-b135-1aad114852fa.png)

Reply:
Writing this comment and ignoring the error saved us one entire line, this already paid off :100: :stuck_out_tongue: